### DISCUSSION TITLE
Option to Not Hash the `salt` parameter

### DISCUSSION THREAD

**Discussion by 0xkowloon** [pcaversaccio/xdeployer#83] [Ideas]
https://github.com/pcaversaccio/xdeployer/pull/82

I think there should be an option to not hash the provided salt because there are cases where the salt is found by a program that finds gas-efficient contract addresses. It wouldn't work if the salt is hashed.

**Comment by pcaversaccio** [comment]
Thanks for opening this discussion - I would like to add an "easy" workaround; you can simply interact with the [`create2deployer`](https://github.com/pcaversaccio/create2deployer/) directly through the block explorer links (or remix) if needed.

So how can you do this using `xdeployer`:
1. Deploy a contract to e.g. a testnet Goerli via `xdeployer`;
2. Copy the init_code from the deployment transaction:

![image](https://user-images.githubusercontent.com/25297591/198589405-09c03d98-0ba9-4115-b76f-53ec3216a302.png)

3. Go to the Etherscan link of the [`create2deployer`](https://github.com/pcaversaccio/create2deployer/) and deploy from there using your already hashed salt:

![image](https://user-images.githubusercontent.com/25297591/198590152-77ffc3cf-c24f-4dde-bd63-632021eb4c59.png)

PS: Of course, you can also retrieve the init_code via ethers.js like that:

```ts
initcode = contract.getDeployTransaction(constructor_args);
```

**Reply by 0xkowloon** [reply]
Thanks for the tip. I ended up writing a Foundry script that calls your contract directly.

```
address internal constant CREATE2_FACTORY = 0x13b0D85CcB8bf860b6b79AF3029fCA081AE9beF2;

function run() public {
  (bool success, bytes memory returndata) = CREATE2_FACTORY.call(
      abi.encodeWithSelector(
          0x66cfa057, // deploy
          0, // value
          SALT,
          type(Contract).creationCode
      )
  );
}
```

**Reply by pcaversaccio** [reply]
ah nice, that's obviously also a great solution :-D
